Handle a few edge cases in the book hotel flow

ArjaanBuijk opened the pull request:
- when trip_destination is not in the database as a hotel location, give a clear message what locations are available.
- handle the user declining a hotel_booking_confirmation

amn41 commented on calm_llm/data/flows/book_hotel.yml:
```diff
@@ -4,7 +4,16 @@ flows:
       - collect: trip_destination
         reset_after_flow_ends: false      
-      - collect: hotel_price_range
+      - id: "verify_hotel_location_is_in_db"
```
wouldn't it be simpler to use a rejection? https://rasa.com/docs/rasa-pro/building-assistants/validating-slot-values#validating-a-slot-values-format-with-a-condition

ArjaanBuijk replied:
Yeah, good question. I thought about that but decided that the current approach follows best practices, for following reasons:

- I considered a `rejection` but the check is too difficult to do it easily via pyred. It needs to handle conversion to lowercase and then check against a range of values. Is probably possible, but I think it will be hard to maintain and debug.

- I considered a slot_validation with a custom action on trip_destination, but that would hide the logic inside a custom action. The trip_destination slot is a global slot, and to check if it is in the database as a hotel location needs to run only for the search_hotel flow. We should not be hiding that type of logic inside a custom action.

amn41 replied:
> We should not be hiding that type of logic inside a custom action

Agreed! This also makes it easier to work in Studio, because the logic is in the flow

ArjaanBuijk commented on calm_llm/data/flows/book_hotel.yml:
```diff
@@ -27,4 +36,12 @@ flows:
         ask_before_filling: true
-      - action: utter_confirm_booked_hotel
+      - noop: true
+        next:
```
Yes, it can be. 
The reason I used this structure to be consistent with the structure that was used after the `collect: hotel_end_date` in the same flow.

I will change both to skip the `noop: true`

ArjaanBuijk replied:
This is now fixed

Balowen commented on calm_llm/domain/book_hotel.yml:
```diff
@@ -28,13 +33,23 @@ slots:
   utter_confirm_booked_hotel:
     - text: Great news! I've successfully booked your stay at the {hotel_name}
+  utter_will_not_book_hotel:
+    - text: Ok, I will not book anything.
```
Won't the pattern_completed run the utter_else resulting in asking "Is there something else I can do for you?" twice?

amn41 replied:
we disabled that pattern in this bot, because there are a lot of multi-skill conversations so it's a bit tedious if you keep getting asked that!

Balowen replied:
Ah. makes sense. Thanks!

ArjaanBuijk commented:
@amn41 , Can you have another look?
- I removed the noop in two places. That nicely cleaned things up
- I added two tests that verify the edge cases handled by these updates
